OneLogin/functions/Set-OneLoginUser.ps1

```diff
@@ -66,9 +66,15 @@ function Set-OneLoginUser
         $username,
 
         [string]
-        $userprincipalname
+        $userprincipalname,
+
+        [int]
+        $state,

```

Comment:
This is nice, but what are your thoughts on implementing separate commands for this? Like `Disable-OneLoginUser`, etc?

Reply:
It might be better to have a new command for the user state (enable, disable, etc.). It would make it easier to use and comment. I also feel that this command would be used on its own more than setting the entire user state. On the other side, it is nice to be able to set a the user paramilitary a one time, either for new users or separated users.

Reply:
Thanks for all of your hard work @mattmcnabb! The `Disable-OneLoginUser` command would be excellent to have. Currently, the only way to "disable" users via the module is to use `Invoke-OneLoginUserLogoff` and `Invoke-OneLoginUserLockout` together. From a sysadmin perspective, the time it would take to disable a bunch of users could be cut down significantly by being able to adjust the user 'status'.

Reply:
Typically, when I disable a user it's for separations. In this case, there are a few things that need to happen. As such, I've addressed this issue by creating a 'Separated' role in OneLogin that disables the account. I use the 'Set-OneLoginUser' command and the role ID to update the user then let OneLogin take care of disabling the account. It's a bit cleaner this way since you can add additional automation to the role/role mappings and any authored user can make the role update.